scripts/todoapp.js

```diff
@@ -106,22 +107,23 @@ class ToDoClass {
     //}
 
     //edit task item
+
     editTask() {
```

Comment:
You need to pass the index of the current task as an argument into editTask(). You can do this using event listeners

The button itself needs to have an index value or key property that is passed as on argument on the onClick  

Resources: https://www.w3schools.com/js/tryit.asp?filename=tryjs_addeventlistener_parameters 

https://stackoverflow.com/questions/16404327/how-to-pass-event-as-argument-to-an-inline-event-handler-in-javascript

Reply:
Hi @xdossantos 
I have now managed to find a solution with linking the edit function to the click event, thanks to one of the links you sent. So now when one clicks on the input field, the corresponding input field becomes editable. I did this by passing the click event as an argument to the editTask function. However, now I'm struggling to get the to-do list to update and save to local storage. I've been trying to find a way of updating the exact corresponding value of the key in the object inside the array and in the JSON local storage string, but I can't seem to get it to work. Do you perhaps have any suggestions?
I also still can't seem to figure out why the date is not converting to a string.